### ISSUE TITLE
`ReferenceError: resetProps is not defined` in Firefox

### ISSUE THREAD

**Issue by akoenig** [jpodwys/superagent-cache#23]
Hey @jpodwys,

thanks for this nice `superagent` plugin. In Chrome everything works like a charm. Unfortunately, I receive a `ReferenceError: resetProps is not defined` when executing my application in Firefox (`43.0.4`).

/André

**Comment by jpodwys** [comment]
Thanks for the report! Could you please paste the code for the query that's having issues? Thanks!

**Comment by akoenig** [comment]
Sure, nothing special. The problem occurs when creating an instance of the cached superagent flavor:

```
import superagentCache from 'superagent-cache';

const request = superagentCache();
```

**Comment by jpodwys** [comment]
If you don't mind, can you try writing this in ES5 syntax? I'm using ES5 and I'm not getting the issue you're reporting in Firefox.

```javascript
var request = require('superagent-cache')();
```

**Comment by jpodwys** [comment]
To clarify, I'm not asking you to ditch ES6, just want to know if that's a starting point for me to track down the issue.

**Comment by jpodwys** [comment]
Any update on this?

**Comment by akoenig** [comment]
Hey @jpodwys, sorry for the late response.

I tried a simple scenario and can spot the error in Firefox as well.

The scenario:

```js
var request = require('superagent-cache')();

console.log(request);
```

I bundled it with `browserify` and executed it in Firefox. Unfortunately, I see the same error.

**Comment by akoenig** [comment]
I guess it correlates with the different hoisting behavior between Firefox and Chrome: https://bugs.chromium.org/p/v8/issues/detail?id=437

**Comment by jpodwys** [comment]
Thanks! I'll try reproducing it again today and see if I can't fix it.

**Comment by jpodwys** [comment]
I'm working on testing this right now. I'm using the latest superagent-cache release (1.3.2) and the latest firefox release (44.0). Please let me know if either of these should change. I know you originally reported the issue with a different version of FF, but I'm guessing the new version doesn't change anything.

**Comment by jpodwys** [comment]
OK, I was able to reproduce the error. The problem for me is that I was using the `uglifyify` package in my app and that was somehow fixing the issue. Once I removed it, I was able to expose the issue. I'm on it now.

**Comment by jpodwys** [comment]
I believe I've fixed the bug. Please try pinning to [this branch](https://github.com/jpodwys/superagent-cache/tree/fixing-firefox-bug) and let me know how it goes. Thanks!

**Comment by jpodwys** [comment]
Because

* this bug is high priority--it prevents any firefox use
* my testing suggests this branch fixes the bug
* I haven't heard back from the reporter yet

I'm merging this.

**Comment by jpodwys** [comment]
Published in `1.3.3`.
